homeassistant/components/openweathermap/__init__.py

```diff
@@ -40,14 +36,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     api_key = entry.data[CONF_API_KEY]
     latitude = entry.data.get(CONF_LATITUDE, hass.config.latitude)
     longitude = entry.data.get(CONF_LONGITUDE, hass.config.longitude)
-    forecast_mode = _get_config_value(entry, CONF_MODE)
     language = _get_config_value(entry, CONF_LANGUAGE)
 
-    config_dict = _get_owm_config(language)
-
-    owm = OWM(api_key, config_dict).weather_manager()
+    owm_client = OWMClient(api_key, lang=language)
     weather_coordinator = WeatherUpdateCoordinator(
-        owm, latitude, longitude, forecast_mode, hass
+        owm_client, latitude, longitude, hass
     )
 
     await weather_coordinator.async_config_entry_first_refresh()
```

Comment:
We should detect that they don't have the v3 api on their account and create an issue in the issue registry telling them what to do.

https://developers.home-assistant.io/docs/core/platform/repairs?_highlight=issue#creating-an-issue

Reply:
i see
i think i can add owm api v2.5 in the lib, then add a switch in HA where you choose which api to use. Then add repair notification to move to 3.0.

Reply:
Sounds good. It should be a config entry option

Reply:
ok, i added support v2.5
added migration, all users will be migrated to onecall v2.5
updated config flow, you can choose between v2.5 and v3.0 (default)
if you choose 2.5, during init repair issue will be created

Reply:
sorry about mess with emails, i will recreate pr later when everything will be done

Reply:
new keys (or rather new accounts) don't work with v2.5 already

Reply:
continue here #116870